### ISSUE TITLE
Panic while parsing git output when git language is not English

### ISSUE THREAD

**Issue by kemenaran** [git-town/git-town#2469]
### Steps to reproduce

```shell
$ LANG=fr_FR git status
Sur la branche improve-setup
Votre branche est en avance sur 'origin/improve-setup' de 4 commits.
  (utilisez "git push" pour publier vos commits locaux)

rien à valider, la copie de travail est propre

$ LANG=fr_FR git town sync
panic: cannot determine the sync status for Git remote "[origin/improve-setup: en avance de 4] Gemfile: declare usage on Darwin 22" and branch name "improve-setup"
```

### Cause

From what I understand of the code, `git-town` attempts to parse the branch status, but doesn't find any of the (English) keywords in the git output (because git runs in French).

### Notes

If this is a legitimate issue (and not a misconfiguration on my side), I'm willing to open a PR that ensures git-town invokes git using the English locale.

**Comment by kevgo** [comment]
Thanks for reporting this! Very exciting that there are french Git Town users! 🎉 This seems like a missing feature on Git Town's side. A PR to fix this is very welcome. The easiest way would be to set an environment variable like `LC_ALL=C` (or whatever makes this work on your machine) when `BackendRunner.execute` calls Git in a subshell here: https://github.com/git-town/git-town/blob/f924604221e570ffb8e1c2f3ac4f1014949ac89c/src/subshell/backend_runner.go#L44

This would only change the language to English when Git Town runs Git commands internally to determine the state of the Git repo. The Git commands that are visible to the end user would still be in your native language.

**Comment by kevgo** [comment]
Btw if you are interested in seeing Git Town's own messages to you in French, you could also make a PR with a french version of this file: https://github.com/git-town/git-town/blob/main/src/messages/en.go

**Comment by kevgo** [comment]
FYI your fix for this has been released as [v10.0.0](https://github.com/git-town/git-town/releases/tag/v10.0.0).
